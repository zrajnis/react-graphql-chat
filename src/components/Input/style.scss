.input-container {
  display: inline-block;
  width: 600px;

  .input {
    border: 1px solid $white;
    color: $black;
    display: block;
    font-size: 22px;
    height: 40px;
    line-height: 40px;
    outline: none;
    padding: 0 15px;
    width: 100%;

    &:focus {
      box-shadow: 0 0 10px $white;
    }
  }
}
