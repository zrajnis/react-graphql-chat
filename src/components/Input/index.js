import React from 'react'
import PropTypes from 'prop-types'

import './style.scss'

const Input = ({ id, label, ...rest }) => 
  <div styleName='input-container'>
    <label htmlFor={id}> {label} </label>
    <input id={id} styleName='input' {...rest} />
  </div>

Input.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string,
}

export default Input
