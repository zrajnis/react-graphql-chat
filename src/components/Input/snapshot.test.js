import React from 'react'
import renderer from 'react-test-renderer'

import Input from 'components/Input'

describe('Input component snapshot', () => {
  it('renders the input', () => {
    const tree = renderer.create(
      <Input
        id='textField'
        label='test label'
        type='text'
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
