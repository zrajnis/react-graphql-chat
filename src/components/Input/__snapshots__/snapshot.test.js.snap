// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component snapshot renders the input 1`] = `
<div>
  <label
    htmlFor="textField"
  >
     
    test label
     
  </label>
  <input
    id="textField"
    type="text"
  />
</div>
`;
